test_images.py

```diff
@@ -23,7 +27,32 @@ def read_test_data_csv(fname: str):
     return result
 
 
+def read_node_region_in_frame_csv(fname: str):
+    with open(fname) as f:
+        result = []
+        for (
+            node_region_visible,
+            robot_x,
+            robot_y,
+            heading,
+            node_region_id,
+        ) in csv.reader(f):
+            result.append(
+                (
+                    node_region_visible == "True",
+                    float(robot_x),
+                    float(robot_y),
+                    float(heading),
+                    int(node_region_id),
+                )
+            )
+    return result
```

Comment:
nit: we could return a list comprehension here instead
```suggestion
        return [
            (
                node_region_visible == "True",
                float(robot_x),
                float(robot_y),
                float(heading),
                int(node_region_id),
            )
            for (
                node_region_visible,
                robot_x,
                robot_y,
                heading,
                node_region_id,
            )
            in csv.reader(f)
        ]
```